vault/resource_identity_group_member_entity_ids.go

```diff
@@ -0,0 +1,167 @@
+package vault
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/vault/api"
+	"github.com/terraform-providers/terraform-provider-vault/util"
+)
+
+func identityGroupMemberEntityIdsResource() *schema.Resource {
+	return &schema.Resource{
+		Create: identityGroupMemberEntityIdsUpdate,
+		Update: identityGroupMemberEntityIdsUpdate,
+		Read:   identityGroupMemberEntityIdsRead,
+		Delete: identityGroupMemberEntityIdsDelete,
+
+		Schema: map[string]*schema.Schema{
+			"member_entity_ids": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Description: "Entity IDs to be assigned as group members.",
+			},
+
+			"exclusive": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     true,
+				Description: "Should the resource manage member entity ids exclusively? Beware of race conditions when disabling exclusive management",
+			},
+
+			"group_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "ID of the group.",
+			},
+
+			"group_name": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Name of the group.",
+			},
+		},
+	}
+}
+
+func identityGroupMemberEntityIdsUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*api.Client)
+	id := d.Get("group_id").(string)
+
+	log.Printf("[DEBUG] Updating IdentityGroupMemberEntityIds %q", id)
+	path := identityGroupIDPath(id)
+
+	vaultMutexKV.Lock(path)
+	defer vaultMutexKV.Unlock(path)
+
+	data := make(map[string]interface{})
+	memberEntityIds := d.Get("member_entity_ids").(*schema.Set).List()
+
+	if d.Get("exclusive").(bool) {
```

Comment:
I'm wondering if we should guard populating `data["member_entity_ids"]` here with a check to ensure the identity group is not of type `external`.

Something like: 
```go
resp, err := readIdentityGroup(client, id)
if err != nil {
    return err
}
t, ok := resp.Data["type"]
if ok && t != "external" {
    // rest of logic for d.Get("exclusive").(bool) here
}
```

The reason for this would be to have consistency with the behavior of `external` type groups that don't use the decoupled `vault_identity_group_member_entity_ids` introduced in this PR. 

For example, this config results in zero `member_entity_ids` in `g1` and no errors from `terraform apply`:
```
resource "vault_identity_group" "g1" {
  name = "g1"
  type = "external"
  external_member_entity_ids = false
  member_entity_ids = ["${vault_identity_entity.user.id}"]
}
resource "vault_identity_entity" "user" {
  name = "user"
}
```

This config results in an error from `terraform apply`:
```
resource "vault_identity_group" "g1" {
  name = "g1"
  type = "external"
  external_member_entity_ids = true
}
resource "vault_identity_group_member_entity_ids" "members" {
  exclusive = true
  member_entity_ids = ["${vault_identity_entity.user.id}"]
  group_id = vault_identity_group.g1.id
}
resource "vault_identity_entity" "user" {
  name = "user"
}
```

Error:
```
Error: error updating IdentityGroupMemberEntityIds "478bbe18-8ef8-7488-101a-6e738cf58b4c": Error making API request.

URL: PUT http://127.0.0.1:8200/v1/identity/group/id/478bbe18-8ef8-7488-101a-6e738cf58b4c
Code: 400. Errors:

* member entities can't be set manually for external groups
```

Reply:
I rather let this exposed via Vault.

The problem with the vault_identity_group approach is from the terraform a user thinks the group has these members for identity -- when in reality it ignored the memberentityid's entirely.

Here Vault tells you there is an error.

Reply:
I actually don't have a strong opinion about whether an error should be returned or not in this case. I think it could be fine either way.

My reason for proposing this is for sake of consistency. I'd prefer the behavior for external groups work the same whether you're using `vault_identity_group_member_entity_ids` or `member_entity_ids`. I think having two different experiences could be confusing. I'm also reluctant to change the current behavior of `member_entity_ids` to return an error because it's been around for a while.

For that reason, I'd prefer to add the check. Let me know your thoughts!